qiskit/transpiler/preset_passmanagers/level2.py

```diff
@@ -111,55 +110,37 @@ def _choose_layout_condition(property_set):
         # layout hasn't been set yet
         return not property_set["layout"]
 
-    # 1a. If layout_method is not set, first try a trivial layout
-    _choose_layout_0 = (
-        []
-        if pass_manager_config.layout_method
-        else [
-            TrivialLayout(coupling_map),
-            Layout2qDistance(coupling_map, property_name="trivial_layout_score"),
-        ]
-    )
-    # 1b. If a trivial layout wasn't perfect (ie no swaps are needed) then try using
-    # CSP layout to find a perfect layout
-    _choose_layout_1 = (
-        []
-        if pass_manager_config.layout_method
-        else CSPLayout(coupling_map, call_limit=1000, time_limit=10, seed=seed_transpiler)
-    )
-
-    def _trivial_not_perfect(property_set):
-        # Verify that a trivial layout  is perfect. If trivial_layout_score > 0
-        # the layout is not perfect. The layout is unconditionally set by trivial
-        # layout so we need to clear it before contuing.
-        if property_set["trivial_layout_score"] is not None:
-            if property_set["trivial_layout_score"] != 0:
-                property_set["layout"]._wrapped = None
-                return True
-        return False
-
-    def _csp_not_found_match(property_set):
-        # If a layout hasn't been set by the time we run csp we need to run layout
+    def _vf2_match_not_found(property_set):
+        # If a layout hasn't been set by the time we run vf2 layout we need to
+        # run layout
         if property_set["layout"] is None:
             return True
-        # if CSP layout stopped for any reason other than solution found we need
-        # to run layout since CSP didn't converge.
+        # if VF2 layout stopped for any reason other than solution found we need
+        # to run layout since VF2 didn't converge.
         if (
-            property_set["CSPLayout_stop_reason"] is not None
-            and property_set["CSPLayout_stop_reason"] != "solution found"
+            property_set["VF2Layout_stop_reason"] is not None
+            and property_set["VF2Layout_stop_reason"].value != "solution found"
         ):
             return True
         return False
 
-    # 1c. if CSP layout doesn't converge on a solution use layout_method (dense) to get a layout
+    # 1a. Try using VF2 layout to find a perfect layout
+    vf2_seed = seed_transpiler
+    if vf2_seed is None:
+        vf2_seed = -1
+    _choose_layout_0 = (
+        [] if pass_manager_config.layout_method else VF2Layout(coupling_map, seed=vf2_seed)
```

Comment:
We can add an extra argument `call_limit` in VF2Layout (similar to CSPLayout) since retworkx supports it https://qiskit.org/documentation/retworkx/stubs/retworkx.vf2_mapping.html#retworkx.vf2_mapping. 
I forgot to mention it in the original PR that added VF2Layout, but we should probably add it here, especially since VF2Layout makes its way to preset passmanagers.
But now comes the question what's a good default value? Honestly, i have no idea... we might have to do some experiments to get a better feeling.